### ISSUE TITLE
Implement direct multi-horizon Ridge net-load forecaster

### ISSUE THREAD

**Issue by jesuscontrerasocana** [jesuscontrerasocana/scaffold#12]
## Goal

Implement a fast, leakage-safe Ridge forecaster for `grid_net_kw` that predicts the full 132-step / 33-hour horizon **directly**, without recursive forecasting.

This replaces the recursive Ridge approach from issue #9. Do **not** reuse the recursive implementation.

The key design is:

```text
information known at decision time t
    ├── model for +15 min  -> grid_net_kw[t + 15 min]
    ├── model for +30 min  -> grid_net_kw[t + 30 min]
    ├── ...
    └── model for +33 h    -> grid_net_kw[t + 33 h]
```

Use one small Ridge model per lead time (132 models total). Ridge is cheap to fit and predict, and this avoids recursive error propagation and the heavy per-step pandas work of the previous implementation.

## Important constraints

- Preserve the existing harness-facing `Forecaster` interface.
- Forecast target is `grid_net_kw`.
- Maximum horizon is 132 quarter-hours.
- No recursive prediction.
- No predicted value may be fed back as an input to another lead-time model.
- Do not create DataFrames/reindex/rolling windows 132 times inside `predict()`.
- Prediction should construct the decision-time features once and the target-time/calendar/exogenous features for the requested horizon in bulk.
- Reuse the generic helpers from issue #7 where appropriate:
  - `build_calendar_features()`
  - `build_lag_features()`
  - `build_rolling_features()`
- Keep the implementation straightforward. No model registry, abstract base classes, factories, feature-class hierarchy, YAML feature configuration, or custom forecasting framework.
- Use sklearn `Ridge(alpha=1.0)` and `StandardScaler`.
- Persist and reload the trained model(s).
- Legacy artifacts from the scrapped recursive Ridge branch do not need to load.

## Forecast formulation

For each lead `h = 1..132`, train a separate model:

```text
X(t, h) -> grid_net_kw(t + h)
```

The row represents a forecast decision made at time `t` for target time `t + h`.

### Decision-time features

These must use only measurements available **strictly before the forecast decision time**.

Initial feature set:

- `lag_15min`: latest known net load, i.e. `grid_net_kw[t - 15 min]`
- `lag_1h`: `grid_net_kw[t - 1 h]`
- `rolling_mean_1h`: mean net load over the 1 hour ending at `t - 15 min`
- `rolling_mean_4h`: mean net load over the 4 hours ending at `t - 15 min`

Build these once per decision timestamp. They are the same recent-state information for all 132 lead-time models.

Do **not** define these relative to the target timestamp. For example, a +24 h forecast may not use `grid_net_kw[target - 15 min]`, because that value is unknown at decision time.

### Target-time features

These describe the timestamp being forecast (`t + h`):

- target time-of-day, encoded as sine and cosine
- target day-of-week dummies
- `most_recent_load_factor_forecast` at the target timestamp

Use `build_calendar_features()` to obtain raw calendar values, then apply the Ridge-specific sin/cos and dummy encoding locally.

`most_recent_load_factor_forecast` is known-future information supplied by the harness and may be used for the corresponding target timestamp.

### Features intentionally excluded initially

Do not add these unless a later issue asks for them:

- month / seasonal month feature
- weekend if day-of-week dummies are already present
- recursive predictions
- direct target-relative lags that would not be known at decision time
- prices
- PV/site-load decomposition
- HGBR

## Training-data construction

Training feature construction should be vectorized over the historical series.

Conceptually:

1. Build decision-time lag/rolling features for all possible decision timestamps.
2. For each lead `h`:
   - target is `grid_net_kw.shift(-h)` aligned to decision time;
   - target timestamp is decision timestamp + `h * 15 min`;
   - build target calendar features for those target timestamps;
   - align the known-future load-factor forecast from the target timestamp;
   - drop only rows for which the required inputs or target are unavailable;
   - fit the model for that lead.

Avoid a Python loop over every training row. A loop over the 132 lead times is fine.

## Scaling

Ridge requires feature scaling because the continuous features have very different magnitudes.

Keep scaling leakage-safe:

- fit scaling only on training data;
- persist the fitted scaler(s);
- reuse the same transformation at inference.

For simplicity, it is acceptable to scale all final numeric input columns, including calendar/dummy columns, if doing so substantially simplifies the implementation.

Prefer the simplest sklearn representation that keeps fit/predict/persistence clear. Do not add a `ColumnTransformer` solely to avoid scaling dummy variables.

## Feature selection

Do **not** use `SequentialFeatureSelector` in this implementation.

The candidate feature set is small and already motivated. Running feature selection independently for 132 models would add substantial training complexity and runtime for little benefit.

Use the full initial feature set for every lead-time model. Ridge regularization already helps with correlated predictors.

If we later want feature selection, treat it as a separate experiment/issue.

## Prediction performance

`predict()` must be fast enough to use inside the rolling forecast evaluator.

At inference for one decision time:

1. Calculate the recent-state lag/rolling features once from `history` strictly before `at_time`.
2. Build calendar features for the requested target `index` in bulk.
3. Read the corresponding `most_recent_load_factor_forecast` values from `future_exog` in bulk.
4. Assemble the feature matrix for the requested horizon.
5. Apply the corresponding lead-time models.

A simple loop over at most 132 already-built feature rows/models is acceptable, but there should be **no repeated pandas history slicing/reindexing/rolling calculations inside that loop**.

The implementation should remain comfortably usable when `scripts/evaluate_forecast.py` calls it thousands of times in a monthly rolling evaluation.

## Missing values

Training:

- drop rows where required historical features, future load-factor forecast, or target are unavailable;
- report how many rows were used per lead or a concise aggregate/range if per-lead detail is cumbersome.

Inference:

- if recent historical lag/rolling inputs are missing, use training-derived median fill values;
- if the known-future load-factor forecast is missing for a target timestamp, use its training median as a fallback rather than failing the whole horizon;
- persist fill values.

Keep this conservative and explicit. Do not infer or rewrite source telemetry.

## Validation and metrics

Do not implement another rolling backtester inside the model. `scripts/evaluate_forecast.py` is the authoritative evaluation of the actual multi-horizon forecast behavior.

The model may expose simple fit diagnostics such as training MAE/RMSE, but do not build a complicated internal holdout framework.

If validation statistics are included inside the model, label them clearly according to what they measure and keep the implementation simple. Do not present a metric as recursive/operational forecast performance when it is not.

## Inspectability

The trained model should be easy to inspect without reverse-engineering `joblib`.

Persist human-readable metadata in `forecaster.json`, including at least:

- `model_name`
- maximum horizon / number of lead models
- feature names in model order
- `alpha`
- fill values
- number of training samples, preferably summarized across leads
- training MAE/RMSE, preferably summarized by lead or with aggregate + selected representative leads
- coefficients and intercepts in a human-readable form

Because there are 132 models, avoid dumping an unreadable wall of data if a concise structure works better. A reasonable representation is a mapping keyed by lead step or a compact list with entries such as:

```json
{
  "lead_steps": 1,
  "lead_minutes": 15,
  "coefficients": {"lag_15min": 1.2, "rolling_mean_4h": 0.7},
  "intercept": 3.1,
  "train_mae": 4.2,
  "train_rmse": 7.1
}
```

The persisted sklearn objects remain authoritative for prediction; JSON inspection data may intentionally duplicate coefficients/metrics.

A `summary()` method is welcome if it remains simple.

## Persistence

Suggested model directory:

```text
models/ridge/
  forecaster.json
  model.joblib
```

`model.joblib` may contain the collection of 132 fitted sklearn pipelines/models.

`Forecaster.save()` / `Forecaster.load()` must preserve predictions across a save/load round trip.

## Tests

Add focused tests covering at least:

1. **Train and predict full horizon**
   - fit the Ridge model;
   - request 132 steps;
   - get 132 finite `net_kw` predictions.

2. **No future realized net-load leakage**
   - predictions must be identical if rows at or after `at_time` in an oversized history frame are replaced with absurd values;
   - the model must only use history strictly before `at_time`.

3. **Long-horizon directness**
   - changing the prediction at an earlier lead must not be able to affect a later lead;
   - there should be no working recursive history / predicted-value feedback path.

4. **Target-time feature alignment**
   - verify that target calendar and load-factor forecast values correspond to `t + h`, not decision time `t`.

5. **Historical feature semantics**
   - verify that `lag_15min`, `lag_1h`, and rolling features are relative to decision time, not target time.

6. **Save/load round trip**
   - loaded model produces identical predictions;
   - human-readable metadata is present.

7. **Missing-value fallback**
   - missing historical feature or future load-factor value uses persisted fill value and still returns a finite prediction.

8. **Performance regression guard**
   - do not add a brittle wall-clock threshold;
   - instead, structure/test the prediction path so lag/rolling feature construction happens once per forecast call, not once per horizon step.

## Acceptance criteria

- [ ] Direct multi-horizon Ridge implemented with one model per lead step, up to 132 steps.
- [ ] No recursive forecasting or predicted-value feedback.
- [ ] Decision-time lags/rolling features are leakage-safe and computed relative to the forecast decision time.
- [ ] Target-time calendar and load-factor features are correctly aligned to each forecasted timestamp.
- [ ] Existing generic feature helpers are reused for bulk feature construction where appropriate.
- [ ] Training is vectorized over rows; only the lead-time loop is expected.
- [ ] Prediction computes historical state features once and does not perform expensive pandas feature reconstruction 132 times.
- [ ] Full sensible feature set is used; no Sequential Feature Selector.
- [ ] Ridge scaling is leakage-safe and persisted.
- [ ] Missing values have explicit persisted median fallbacks.
- [ ] `forecaster.json` contains useful human-readable model metadata, including coefficients/intercepts or an equivalent inspectable representation.
- [ ] `model.joblib` persists the fitted models/scalers.
- [ ] Save/load preserves predictions.
- [ ] Harness-facing `Forecaster` interface remains compatible.
- [ ] Focused tests cover leakage, direct-horizon semantics, feature alignment, persistence, missing values, and full 132-step prediction.
- [ ] No unnecessary architecture or abstractions are introduced.

## Out of scope

- HGBR
- load/PV decomposition
- probabilistic/scenario forecasting
- hyperparameter tuning
- recursive forecasting
- monthly peak optimization
- changes to the optimizer
- custom model/feature framework
